tests/event-log/event-stream.spec.ts

```diff
@@ -57,45 +77,80 @@ describe('EventStream', () => {
+      const { message } = event;
+      const messageCid = await Message.getCid(message);
+      sub1MessageCids.push(messageCid);
+    };
 
-  it('does not emit messages if subscription is closed', async () => {
-    const messageCids: string[] = [];
-    const handler = async (_tenant: string, event: MessageEvent, _indexes: KeyValues): Promise<void> => {
+    const sub2MessageCids: string[] = [];
+    const handler2 = async (_tenant: string, event: MessageEvent, _indexes: KeyValues): Promise<void> => {
       const { message } = event;
       const messageCid = await Message.getCid(message);
-      messageCids.push(messageCid);
+      sub2MessageCids.push(messageCid);
     };
-    const subscription = await eventStream.subscribe('did:alice', 'sub-1', handler);
+
+    const subscription1 = await eventStream.subscribe('did:alice', 'sub-1', handler1);
+    const subscription2 = await eventStream.subscribe('did:alice', 'sub-2', handler2);
 
     const message1 = await TestDataGenerator.generateRecordsWrite({});
     const message1Cid = await Message.getCid(message1.message);
     eventStream.emit('did:alice', { message: message1.message }, {});
-    await subscription.close();
+
+    // Use the TimingUtils to poll until the expected results are met
+    await TestTimingUtils.pollUntilSuccessOrTimeout(async () => {
+      expect(sub1MessageCids).to.have.length(1);
+      expect(sub1MessageCids).to.have.members([ message1Cid ]);
+
+      expect(sub2MessageCids).to.have.length(1);
+      expect(sub2MessageCids).to.have.members([ message1Cid ]);
+    });
+
+    await subscription1.close(); // close subscription 1
 
     const message2 = await TestDataGenerator.generateRecordsWrite({});
+    const message2Cid = await Message.getCid(message2.message);
     eventStream.emit('did:alice', { message: message2.message }, {});
 
-    await Time.minimalSleep();
+    // Use the TimingUtils to poll until the expected results are met
+    await TestTimingUtils.pollUntilSuccessOrTimeout(async() => {
+      // subscription 2 should have received the message
+      expect(sub2MessageCids.length).to.equal(2);
+      expect(sub2MessageCids).to.have.members([ message1Cid, message2Cid]);
 
-    expect(messageCids).to.have.members([ message1Cid ]);
+      // subscription 1 should not have received the message
+      expect(sub1MessageCids).to.have.length(1);
+      expect(sub1MessageCids).to.have.members([ message1Cid ]);
+    });
+
+    await subscription2.close();
   });
 
-  it('does not emit messages if event stream is closed', async () => {
-    const messageCids: string[] = [];
-    const handler = async (_tenant: string, event: MessageEvent, _indexes: KeyValues): Promise<void> => {
-      const { message } = event;
-      const messageCid = await Message.getCid(message);
-      messageCids.push(messageCid);
+  it('does not receive messages if event stream is closed', async () => {
```

Comment:
I'm considering removing this test completely. I don't like the hack here, but can't think of a better way to test it here.

We could implement this test in each concrete implementations of the EventStream.

Reply:
I am okay for us to remove this if the code paths are covered by other  tests.

Alternatively, you could change the override in `test-suite.ts` from
`eventStream?: EventStream` -> `eventStreamCreator?: () => EventStream`

And create an addition `eventStream` in this test for control.

Reply:
I think removing it is the best bet, it's tested for `EventEmitterStream` to make sure there are no more event listeners.